docs/certificate-specification.md

```diff
@@ -0,0 +1,112 @@
+# Certificate Specification
+
+This document includes the requirements for root, intermediate, and issued certificates.
+
+## Root Certificate
+
+A root certificate MUST:
+
+* Specify a Subject with a common name and organization
+* Specify an Issuer with the same values as the Subject
+* Specify Key Usages for Certificate Sign and CRL Sign
+* Specify Basic Constraints to `CA:TRUE`
+* Specify a unique, positive, 160 bit serial number
+* Specify a Subject Key Identifier
+* Be compliant with [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280)
+
+A root certificate MUST NOT:
+
+* Specify other Key Usages besides Certificate Sign and CRL Sign
+* Specify any Extended Key Usages
+
+A root certificate SHOULD:
+
+* Use the signing algorithm ECDSA NIST P-384 (secp384r1) or stronger, or RSA-4096
+* Have a lifetime that does not require frequent rotation, such as 10 years
+
+A root certificate MAY:
+
+* Specify a Basic Constraints path length constraint to prevent additional CA certificates
+  from being issued beneath the root
+* Specify an Authority Key Identifier. If specified, it MUST be the same as the Subject Key Identifier
+* Specify other values in the Subject.
+
+## Intermediate Certificate
+
+An intermediate certificate MUST:
+
+* Specify a Subject with a common name and organization
+* Specify an Issuer equal to the parent certificate's Subject
+* Specify Key Usages for Certificate Sign and CRL Sign
+* Specify an Extended Key Usage for Code Signing
+* Specify a lifetime that does not exceed the parent certificiate
+* Specify Basic Constraints to `CA:TRUE`
+* Specify a unique, positive, 160 bit serial number
+* Specify a Subject Key Identifier
+* Specify an Authority Key Identifier equal to the parent certificate's Subject Key Identifier 
+* Be compliant with [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280)
+
+An intermediate certificate MUST NOT:
+
+* Specify other Key Usages besides Certificate Sign and CRL Sign
+* Specify other Extended Key Usages besides Code Signing
+
+An intermediate certificate SHOULD:
+
+* Specify a Basic Constraints path length constraint of 0, `pathlen:0`. This limits the intermediate
+  CA to only issue end-entity certificates
+* Use the signing algorithm ECDSA NIST P-384 (secp384r1) or stronger, or RSA-4096
+* Have a lifetime that does not require frequent rotation, such as 3 years
+
+An intermediate certificate SHOULD NOT:
+
+* Use a different signature scheme (ECDSA vs RSA) than its parent certificate, as some clients do not support this
+
+## Issued Certificate
+
+An issued certificate MUST:
+
+* Specify a Subject Alternative Name as a critical extension. It MUST be populated by either:
```

Comment:
Just to make this as explicit as possible (it's also listed under `MUST NOT`, but I figured a little bit of redundancy doesn't hurt):

```suggestion
* Specify exactly one Subject Alternative Name, as a critical extension. It MUST be populated by either:
```